index.js

```diff
@@ -48,50 +48,257 @@ const player = {
+    }
+  }
 }
 
+
 function addSong(title, album, artist, duration, id) {
-  // your code here
+  let newID=0;
+  let max=1;
+  if(id){
+  for (let index = 0; index < player.songs.length; index++) {
+    if(player.songs[index].id===id){
+      throw new Error("Sorry , this ID already taken") }}}
+    else {
+  for (let i = 0; i < player.songs.length; i++) {
+    if(player.songs[i].id>max){
+      max === player.songs[i].id;}
+  }
+  newID=max+1;
+} 
+  duration = duration.split(":");
+  duration = parseInt(duration[0] *60) + parseInt(duration[1]);
+  let newSong= {
+    id: id,
+    title: title,
+    album: album,
+    artist: artist,
+    duration: duration };
+    player.songs.push(newSong);
+    return newID;
 }
 
+
+function getPlaylistByID(id){
+  for (let index = 0; index < player.playlists.length; index++) {
+    if(player.playlists[index].id === id){
+      return player.playlists[index];
+    }
+}
+throw new Error("This playlist are not exist")
+}
+
+
 function removePlaylist(id) {
-  // your code here
+  let playlistIndex=getPlaylistByID(id);
+  player.playlists.splice(playlistIndex,1);
 }
 
+
 function createPlaylist(name, id) {
-  // your code here
+  let newP_ID=0
+  let maxP_ID=1
+  if(id){
+  for (let index = 0; index < player.playlists.length; index++) {
+    if(player.playlists[index].id===id){
+      throw new Error("Sorry , this ID already taken")
+    }
+ }
+ }
+else {
+  for (let i = 0; i < player.playlists.length; i++) {
+    if(player.playlists[i].id>maxP_ID){
+      maxP_ID === player.playlists[i].id;
+    }
+}
+newP_ID=maxP_ID;
+}
+  
+  let newPlaylist={
+    id:id,
+    name:name,
+    songs:[]
+  }
+  player.playlists.push(newPlaylist);
+  return newP_ID;
 }
 
+
 function playPlaylist(id) {
-  // your code here
+  let Pindex=getPlaylistByID(id);
+  for (let i = 0; i < Pindex.songs.length; i++) { 
+    playSong (Pindex.songs[i]);  
+  }
 }
 
+
+function indexSongInPlaylist( playlistId,songId) {
+  let playlist = getPlaylistByID(playlistId)
+  for (let index = 0; index < playlist.songs.length; index++) {
+    if (playlist.songs[index] == songId)
+      return index;
+  }
+  return -1;
+}
+
+
+function songExist(songId){
+  for(let i=0;i<player.songs.length;i++){
+    if(player.songs[i].id===songId){
+      return true;
+    }
+  }
+  return false;
+}
+
+
+function playlistExist(id) {
+  for (let i = 0; i < player.playlists.length; i++) {
+    if (player.playlists[i].id == id)
+      return true
+  }
+  return false
+}
+
+
 function editPlaylist(playlistId, songId) {
-  // your code here
+  let songIndex = indexSongInPlaylist(playlistId, songId)
+  let playlist = getPlaylistByID(playlistId)
+  if(songExist(songId) && playlistExist(playlistId)){
+    if(songIndex == -1){
+      playlist.songs.push(songId);
+    }
+    else if(playlist.songs.length > 1){
+      playlist.songs.splice(songIndex,1)
+    }
+    else{
+      removePlaylist(playlist.id)
+    }
+  }
+  else throw new Error("Somthing wrong");
```

Comment:
This error is too vague. Imagine you are using this function and get this error. It doesn't give you any clue of what actually went wrong.